lib/buffer-search.js

```diff
@@ -224,6 +224,8 @@ class BufferSearch {
       }
 
       const newMarkers = this.createMarkers(scanStart, scanEnd);
+      if(newMarkers === false || !newMarkers.length) return; // Invalid/missing find pattern or no new markers
```

Comment:
Even if there are no *new* markers, we still might want to destroy some old markers, and we certainly want to continue processing the other entries in `changes`, because they might cause us to find new markers.

For example, suppose your editor was in this configuration (cursors represented by `|`):

```
1.   one
2.   two|
3.   three
4.   foo|ur
```

and you were searching for the pattern `two|four` and typed a `backspace`. That would *destroy* the match on line 2 but *create* a match on line 4.

In that situation, we wouldn't to return early from `bufferStoppedChanging` just because we didn't find any new markers in the first changed region.